pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -169,11 +169,11 @@ func (o *OpenIDCProvider) SearchPrincipals(searchValue, principalType string, to
 	}
 
 	p := v3.Principal{
-		ObjectMeta:    metav1.ObjectMeta{Name: o.GetName() + "_" + principalType + "://" + searchValue},
+		ObjectMeta:    metav1.ObjectMeta{Name: o.Name + "_" + principalType + "://" + searchValue},
 		DisplayName:   searchValue,
 		LoginName:     searchValue,
 		PrincipalType: principalType,
-		Provider:      o.GetName(),
+		Provider:      o.Name,
 	}
```

Comment:
Would be nice to re-use the `toPrinciples` methods below but might be too complicated here since it looks like you are using both and you don't have necessary info. 
Unrelated to this PR, but I am confused why you just return this directly with the search value included, its just a placeholder ?

Reply:
I think this is because  OIDC does not support searching?

Reply:
Correct - This is how our SAML providers are currently handling `SearchPrincipals` because they do not support search functionality for the various SAML providers.

The other option would be a `return nil, errors.New("[generic oidc]: not implemented")` instead of implementing it.